internal/provider/newrelic/provider_integration_test.go

```diff
@@ -168,12 +181,228 @@ func Test_Getting_external_metric_through_proxy(t *testing.T) {
 	})
 }
 
-func newrelicProviderWithMetric(t *testing.T, metric newrelic.Metric) provider.ExternalMetricsProvider {
-	t.Helper()
+// As The server must be run on port 3000 to be run locally as per the new relic client requirements, subtests
+// cannot be run in parallel.
+//
+// For more details check https://github.com/newrelic/newrelic-client-go/blob/main/pkg/region/region_constants.go
+//
+//nolint:funlen,tparallel
+func Test_does_not_hide_backend_errors(t *testing.T) {
+	t.Parallel()
 
-	testEnv := testutil.TestEnv{}
+	ctx := testutil.ContextWithDeadline(t)
+
+	testEnv := testutil.TestEnv{
+		StartKubernetes: true,
+		Region:          "Local",
+	}
 	testEnv.Generate(t)
 
+	url := fmt.Sprintf("%s/apis/external.metrics.k8s.io/v1beta1/namespaces/*/test_metric", testEnv.BaseURL)
+
+	p := newrelicProviderWithMetric(t, newrelic.Metric{
+		Query: testIntegrationQuery,
+	}, &testEnv)
+
+	runAdapter(t, &testEnv, p)
+
+	//nolint:paralleltest
+	t.Run("when_response_error", func(t *testing.T) {
+		cases := map[string]struct {
+			handlerFunc       func(w http.ResponseWriter, r *http.Request)
+			wantErr           bool
+			makeExtraRequests bool
+		}{
+			"is_200_with_empty_data": {
+				handlerFunc: func(w http.ResponseWriter, _ *http.Request) {
+					w.WriteHeader(http.StatusOK)
+				},
+				wantErr: true,
+			},
+			"is_400": {
+				handlerFunc: func(w http.ResponseWriter, _ *http.Request) {
+					w.WriteHeader(http.StatusBadRequest)
+				},
+				wantErr: true,
+			},
+			"is_401": {
+				handlerFunc: func(w http.ResponseWriter, _ *http.Request) {
+					w.WriteHeader(http.StatusUnauthorized)
+				},
+				wantErr: true,
+			},
+			"is_403": {
+				handlerFunc: func(w http.ResponseWriter, _ *http.Request) {
+					w.WriteHeader(http.StatusForbidden)
+				},
+				wantErr: true,
+			},
+			"is_500": {
+				handlerFunc: func(w http.ResponseWriter, _ *http.Request) {
+					w.WriteHeader(http.StatusInternalServerError)
+				},
+				wantErr: true,
+			},
+			"is_200_for_first_requests_and_then_500": {
+				handlerFunc: func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("Content-Type", "application/json")
+					w.WriteHeader(http.StatusOK)
+					_, _ = w.Write([]byte(`{"data": {
+				"actor": {
+				  "account": {
+					"nrql": {
+					  "results": [{"one": 0.015}]
+					}
+				  }
+				}
+			}}`))
+				},
+				wantErr:           false,
```

Comment:
i found it ambiguous, because we want to not error and the error. I would do this in a separate test and leave this table driven test for a simpler request.

Reply:
Separated in another subtest to simplify logic here https://github.com/newrelic/newrelic-k8s-metrics-adapter/pull/52/commits/0fff43e9a28e9f4545222e9f85adecff3a15c642

Reply:
Separated in another subtest to simplify logic here 0e6bfd2c81bd54f0fac543e3094fd3f1d445e536